crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -83,9 +88,42 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         visitor.raises
     };
 
+    if raises.is_empty() {
+        return;
+    }
+
+    let handled_exceptions = helpers::extract_handled_exceptions(handlers);
+    let comparables: Vec<ComparableExpr> = handled_exceptions
+        .iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { .. }) = handler {
```

Comment:
Do we need this condition? I kind of suspect we can omit it entirely, to handle cases like:

```
try:
  raise foo.Bar()
except foo.Bar:
  pass
```

Right now, this only seems to support names, but `ComparableExpr` should make it such that we can support any expression.

Reply:
@charliermarsh I think you're right. Thanks for the pointers - need to get more familiar with the AST side of things. 

I've given you edit permissions, I'm going to be pretty busy for the next month or so and am not sure I'll be able to monitor this. Feel free to make any changes you need. Thanks again